lib/internal/process/promises.js

```diff
@@ -25,14 +25,24 @@ const pendingUnhandledRejections = [];
 const asyncHandledRejections = [];
 let lastPromiseId = 0;
 
-const states = {
-  none: 0,
-  warn: 1,
-  strict: 2,
-  default: 3
-};
-
-let state;
+// --unhandled-rejection=none:
+// Emit 'unhandledRejection', but do not emit any warning.
+const kIgnoreUnhandledRejections = 0;
+// --unhandled-rejection=warn:
+// Emit 'unhandledRejection', then emit 'UnhandledPromiseRejectionWarning'.
+const kAlwaysWarnUnhandledRejections = 1;
+// --unhandled-rejection=strict:
+// Emit 'uncaughtException'. If it's not handled, print the error to stderr
+// and exit the process.
+// Otherwise, emit 'unhandledRejection'. If 'unhandledRejection' is not
```

Comment:
TBH this behavior looks weird to me. It contradicts with the description in `test/parallel/test-promise-unhandled-error.js`:

```
// Verify that unhandled rejections always trigger uncaught exceptions instead
// of triggering unhandled rejections.
```

Even though that test specifically tests:

https://github.com/nodejs/node/blob/282e2f669c0e4738a13c6b49d59dbd83002f3953/test/parallel/test-promise-unhandled-error.js#L37

Reply:
cc @BridgeAR

Reply:
The test description is faulty. The original implementation changed multiple times and it seems like the description was not updated appropriately.

The original intention was to replace the unhandled rejections with the exception. Later on it was decided that the current unhandled rejections hook behavior should not be influenced by the flag at all.